### ISSUE TITLE
[BUG] Pennylane import fails on Google Colab

### ISSUE THREAD

**Issue by ChaakuDaaku** [PennyLaneAI/pennylane#2547]
### Expected behavior

After installing pennylane through pip on Google Colab, the `import pennylane as qml` should work smoothly.

### Actual behavior

The import fails and throws a `torch.linalg.eigh` `KeyError` as shown in the traceback below:

### Additional information

This occurs everytime on a new Google Colab notebook.

The only way to make it work is to install torch LTS(1.8.2) with Cuda 11.1 version.
`pip3 install torch==1.8.2+cu111 torchvision==0.9.2+cu111 torchaudio==0.8.2 -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html`

### Tracebacks

```shell
KeyError                                  Traceback (most recent call last)
<ipython-input-2-65e0d8c219c8> in <module>()
----> 1 import pennylane

5 frames
/usr/local/lib/python3.7/dist-packages/pennylane/math/single_dispatch.py in <module>()
    328         # autoray.py:84: UserWarning: torch.symeig is deprecated in favor of torch.linalg.eigh
    329         # and will be removed in a future PyTorch release.
--> 330         del ar.autoray._FUNCS["torch", "linalg.eigh"]
    331 except ImportError:
    332     pass

KeyError: ('torch', 'linalg.eigh')
```

### System information

```shell
python --version
Python 3.7.13
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by josh146** [comment]
Hi @ChaakuDaaku! This is due to a new release of Autoray, which PennyLane is not compatible with. We are working fast to fix this issue - in the meantime, if you downgrade Autoray to version 0.2.5, this should fix the bug:

```
pip install autoray==0.2.5
```

**Comment by antalszava** [comment]
Hi @ChaakuDaaku, we've released v0.23.1 of PennyLane which should resolve this behaviour. I'll leave this issue open to see if all's good on your end too. :+1:

**Comment by ChaakuDaaku** [comment]
Hi @antalszava ! Yep the issue seems to be resolved. Thank you so much!
You can close the issue 👍
